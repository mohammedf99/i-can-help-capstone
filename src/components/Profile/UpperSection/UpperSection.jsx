import React, { useContext, useEffect, useState } from "react";
import { Row, Col } from "antd";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMapMarkerAlt, faUser } from "@fortawesome/free-solid-svg-icons";
import Link from "next/link";
import {
  UpperSectionCont,
  MyPostsCard,
  PinnedPostCard,
} from "./UpperSection.styled";
import DataContext from "../../../Utilities/Contexts/DataContext";
import AuthContext from "../../../Utilities/Contexts/AuthContext";
import { getUser } from "../../../Utilities/FirebaseUtilities";

const UpperSection = () => {
  const [userPosts, setUserPosts] = useState({});
  const data = useContext(DataContext).userData;
  const { currentUser } = useContext(AuthContext);
  useEffect(() => {
    getUser(currentUser?.uid).then((user) => {
      user?.ref
        .collection("posts")
        .onSnapshot((snapshot) =>
          setUserPosts((prev) => ({ ...prev, myPosts: snapshot.docs.length }))
        );
    });
  }, [currentUser]);

  return (
    <UpperSectionCont>
      <div className="left-section">
        <div className="img-div">
          <img src={data?.picture} alt="user" />
        </div>
        <div className="user-info-div">
          <h1 className="name">{data?.name}</h1>
          <h2 className="job">{data?.job}</h2>
          <h3 className="location">
            <span className="location-icon">
              <FontAwesomeIcon icon={faMapMarkerAlt} />
            </span>
            {data?.location}
          </h3>
        </div>
      </div>

      <div className="right-section">
        <MyPostsCard>
          <div className="card-upper">
            <h3 className="post-number">{userPosts.myPosts}</h3>
          </div>
          <div className="card-lower">
            <h3 className="post-title">
              <Link href="/myPosts">My posts</Link>
            </h3>
          </div>
        </MyPostsCard>
        <PinnedPostCard>
          <div className="card-upper">
            <h3 className="post-number">{data?.pinnedPosts?.length || 0}</h3>
          </div>
          <div className="card-lower">
            <h3 className="post-title">
              <Link href="/pinnedPosts">Pinned posts</Link>
            </h3>
          </div>
        </PinnedPostCard>
      </div>
    </UpperSectionCont>
  );
};

export default UpperSection;
