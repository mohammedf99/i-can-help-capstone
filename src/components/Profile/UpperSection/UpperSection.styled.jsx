import styled from "styled-components";

export const ProImgDiv = styled.div`
  width: 150px;
  height: 150px;
  border-radius: 50%;
  background-image: url("https://cdn.icon-icons.com/icons2/2506/PNG/512/user_icon_150670.png");
  background-color: #1c1259;

  img {
    width: 150px;
    height: 150px;
    object-fit: cover;
    border-radius: 50%;
  }
`;

export const ProUserName = styled.h2`
  color: #1c1259;
  font-size: 28px;
  font-weight: 800;
  margin: 0;
`;

export const WorkTitle = styled.h3`
  color: #1c1259;
  font-size: 24px;
  font-weight: 600;
  margin: 0;
`;

export const LocationText = styled.h4`
  color: #1c1259;
  font-size: 18px;
  font-weight: 600;
  margin: 0;
  display: inline-block;
  padding-left: 10px;
`;

export const MyPostsDiv = styled.div`
  box-sizing: border-box;
  width: 90%;
  height: 140px;
  background-color: #1c1259;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 5px;

  .card-upper {
    height: 60%;
  }

  h2 {
    font-size: 4rem;
    color: #fafaff;
    font-weight: 700;
    padding: 0;
    margin: 0;
  }

  .card-lower {
    height: 40%;
  }

  h3 {
    font-size: 2rem;
    color: #fafaff;
    font-weight: 700;
  }

  .card-upper,
  .card-lower {
    display: flex;
    align-items: center;
  }

  a {
    text-decoration: none;
    color: inherit;
  }

  a:hover {
    opacity: 0.8;
  }
`;

export const PinnedPostsDiv = styled.div`
  box-sizing: border-box;
  width: 90%;
  height: 140px;
  background-color: #FF9B00;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 5px;

  .card-upper {
    height: 60%;
  }

  h2 {
    font-size: 4rem;
    color: #fafaff;
    font-weight: 700;
    padding: 0;
    margin: 0;
  }

  .card-lower {
    height: 40%;
  }

  h3 {
    font-size: 2rem;
    color: #fafaff;
    font-weight: 700;
  }

  .card-upper,
  .card-lower {
    display: flex;
    align-items: center;
  }

  a {
    text-decoration: none;
    color: inherit;
  }

  a:hover {
    opacity: 0.8;
  }
`;

export const numOfPosts = styled.span`
  color: #red;
`;