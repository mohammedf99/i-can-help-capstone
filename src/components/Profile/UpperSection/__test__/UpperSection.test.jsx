import * as React from "react";
import * as renderer from "react-test-renderer";
import UpperSection from "../UpperSection";

it("renders Profile upper section correctly", () => {
  const tree = renderer.create(<UpperSection />).toJSON();
  expect(tree).toMatchSnapshot();
});