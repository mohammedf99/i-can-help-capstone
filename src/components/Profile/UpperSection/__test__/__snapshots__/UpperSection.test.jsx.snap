// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile's upper section correctly 1`] = `
<section>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8"
          style={Object {}}
        >
          <div
            className="UpperSectionstyled__ProImgDiv-sc-1uo74lt-0 eAzPa-d"
          >
            <img
              alt="profile"
              src="https://images.pexels.com/photos/220453/pexels-photo-220453.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
            />
          </div>
        </div>
        <div
          className="ant-col ant-col-16"
          style={Object {}}
        >
          <div
            className="ant-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-24"
              style={Object {}}
            >
              <h2
                className="UpperSectionstyled__ProUserName-sc-1uo74lt-1 diuZhs"
              >
                John Doe
              </h2>
            </div>
            <div
              className="ant-col ant-col-24"
              style={Object {}}
            >
              <h3
                className="UpperSectionstyled__WorkTitle-sc-1uo74lt-2 hcJcZz"
              >
                Gardner
              </h3>
            </div>
            <div
              className="ant-col ant-col-24"
              style={Object {}}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-map-marker-alt fa-w-12 "
                data-icon="map-marker-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "#EE4266",
                  }
                }
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <h4
                className="UpperSectionstyled__LocationText-sc-1uo74lt-3 fcwTUB"
              >
                Erbil, IQ
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-16"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8"
          style={Object {}}
        >
          <div
            className="UpperSectionstyled__MyPostsDiv-sc-1uo74lt-4 cNOGuD post-card"
          >
            <div
              className="card-upper"
            >
              <h2>
                5
              </h2>
            </div>
            <div
              className="card-lowe"
            >
              <h3>
                <a
                  href="#"
                >
                  My posts
                </a>
              </h3>
            </div>
          </div>
        </div>
        <div
          className="ant-col ant-col-8"
          style={Object {}}
        >
          <div
            className="UpperSectionstyled__PinnedPostsDiv-sc-1uo74lt-5 gCVRbc card"
          >
            <div
              className="card-upper"
            >
              <h2>
                5
              </h2>
            </div>
            <div
              className="card-lower"
            >
              <h3>
                <a
                  href="#"
                >
                  Pinned posts
                </a>
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
